metricbeat/module/coredns/stats/stats_integration_test.go

```diff
@@ -0,0 +1,89 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// +build integration
+
+package stats
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/tests/compose"
+	mbtest "github.com/elastic/beats/metricbeat/mb/testing"
+)
+
+func TestData(t *testing.T) {
+	compose.EnsureUp(t, "coredns")
```

Comment:
See my comment above. If you make the change to `testdata` this here will not be needed anymore.

Reply:
Just to mention here, that we use [WriteEventsReporterV2Cond](https://github.com/elastic/beats/blob/45074a3efb74592505f90c9fc70e5dd75d898c1f/metricbeat/mb/testing/data_generator.go#L104) since we we have conditional events. Is this handled with the new suggested approach?

Reply:
The nice part about the new approach is that in the `docs.plain` you can put in only the event / data you want in the `data.json`. But TBH I'm not sure how well this will work with a prometheus output, I would assume it should also work?

Reply:
Will look into it, thanks!